app/src/main/java/com/example/bottomnav/VariableContents.java

```diff
@@ -1,20 +1,121 @@
 package com.example.bottomnav;
 
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTDeclarator;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTEqualsInitializer;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTLiteralExpression;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTNamedTypeSpecifier;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTSimpleDeclSpecifier;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.CPPASTSimpleDeclaration;
+
+import java.util.Optional;
 public class VariableContents {
-    String value;
-    String name;
-    String typeQualifer;
-    String payloadDataType;
-    int packetSize;
-
-    public VariableContents(String iName, String iValue, String iTQ, String iType) {
-        value = iValue;
-        name = iName;
-        typeQualifer = iTQ;
-        payloadDataType = iType;
+    public String name;
+    public String typeQualifier;
+    public String payloadDataType;
+    public Optional<String> value;
+
+
+    public VariableContents(String givenName, String givenTQ,
+                            String givenType, Optional<CPPASTEqualsInitializer> givenInit) {
+        name = givenName;
+        typeQualifier = givenTQ;
+        payloadDataType = givenType;
+        value = getValue(givenInit);
+    }
+
+    public static Optional<VariableContents> getContents(CPPASTSimpleDeclaration dec) throws Exception {
+        if (dec.getDeclSpecifier() instanceof CPPASTSimpleDeclSpecifier) {
+            return makeOptFromDecl((CPPASTSimpleDeclSpecifier) dec.getDeclSpecifier(),
+                    (CPPASTDeclarator) dec.getDeclarators()[0]);
+        } else {
+            return makeOptFromDecl((CPPASTNamedTypeSpecifier) dec.getDeclSpecifier(),
+                    (CPPASTDeclarator) dec.getDeclarators()[0]);
+        }
+    }
+
+    /**
+     *
+     * @param simpleSpec
+     * @param declarator
+     * @return Optional<Components>
+     * @throws Exception
+     *
+     * Difference: Simple specifiers don't quantifier types and do not have name object
+     */
+    private static Optional<VariableContents> makeOptFromDecl(CPPASTSimpleDeclSpecifier simpleSpec,
```

Comment:
```suggestion
    private static Optional<VariableContents> fromDecl(CPPASTSimpleDeclSpecifier simpleSpec,
```
simpler name, especially since there isn't a non-Optional version (and similarly below)